app/models/reservation_frame.rb

```diff
@@ -3,15 +3,36 @@ class ReservationFrame < ApplicationRecord
   belongs_to :time_frame
   has_one :reservation, dependent: :restrict_with_exception
 
-  validates :date, presence: true
-
+  validates :date, presence: true, uniqueness: { scope: %i[time_frame_id planner] }
   scope :not_canceled, -> { where(canceled_at: nil) }
 
+  validate :the_reservation_on_sunday_is_denied
+  validate :saturday_is_shorttime
+  validate :the_date_in_the_past_or_over_a_year_after_today_is_invalid
+
   def reserved?
     reservation.present?
   end
 
   def canceled?
     canceled_at?
   end
+
+  def the_reservation_on_sunday_is_denied
+    return unless date.sunday?
+
+    errors.add(:date, '日曜日はゆっくり過ごしましょう。')
+  end
+
+  def saturday_is_shorttime
+    return unless date.wday == 6 && !time_frame_id.between?(3, 11)
+
+    errors.add(:date, '土曜日は11時〜15時の間で指定してください。')
+  end
+
+  def the_date_in_the_past_or_over_a_year_after_today_is_invalid
```

Comment:
[imo] たとえば自分ならこう命名しますという例です。

```suggestion
  def date_must_be_after_tomorrow_and_within_a_year
```

Reply:
なるほど、ありがとうございます！